docs/src/content/docs/reference/configuration.mdx

```diff
@@ -65,6 +65,33 @@ type LogoConfig = { alt?: string; replacesTitle?: boolean } & (
 
 Configure the table of contents shown on the right of each page. By default, `<h2>` and `<h3>` headings will be included in this table of contents.
 
+### `expressiveCode`
+
+**type:** `StarlightExpressiveCodeOptions | boolean`
+
+Starlight uses [Expressive Code](https://github.com/expressive-code/expressive-code/tree/main/packages/astro-expressive-code) to render code blocks on your site. The integration is enabled by default and can be disabled by passing `false` to this option.
+
+You can customize your code blocks by passing an object with any of the following properties:
+
+- `themes` — an array of themes that should be used to render code blocks. The following item types are supported in this array:
+  - any theme name bundled with Shiki (e.g. `dracula`)
+  - any theme object compatible with VS Code or Shiki (e.g. imported from an NPM theme package)
+  - any ExpressiveCodeTheme instance (e.g. using `ExpressiveCodeTheme.fromJSONString(...)` to load a custom JSON/JSONC theme file yourself)
+
+  Defaults to the dark and light variants of the "Night Owl" theme by Sarah Drasner, which are also exported from `@astrojs/starlight/expressive-code` as `getStarlightDefaultThemes`.
```

Comment:
Instead of exporting this method (I’m assuming for cases where people want to use one of the two default themes?), what do you think of giving these string values people can use?

e.g. `themes: ['starlight-light', 'starlight-dark']` is the default and we resolve those two strings under the hood for people? Seems nicer than exposing the internal function which returns an array it is hard to pick one theme out of.

I think this is also the only place exposing `@astrojs/starlight/expressive-code` is necessary? In which case I think it would be a benefit of this change to avoid exposing that in package `exports`.

Reply:
Yes, I exported these to allow people to reuse our special Starlight versions of the the Night Owl themes in their own set of themes. I like the idea of making them accessible through a string value, and will add that in favor of the `getStarlightDefaultThemes` export.

Exposing `@astrojs/starlight/expressive-code` will still be necessary to allow users deeper customization without risking to run into package version conflicts by having to install their own copy of astro-expressive-code. I mentioned in [this earlier comment](https://github.com/withastro/starlight/pull/742#discussion_r1385148195) what the potential customization use cases are.

Reply:
Aha, got it. Maybe we should make a exports-specific module for the public API? So users can’t accidentally import some of the internal functions that are exported to be used in Starlight’s `index.ts`?

Reply:
Great point! I'll do that and create a separate file for the export.

Reply:
Thank again for helping me understand everything. I think this would then be the last outstanding point to tackle. In the meantime, I’ll also open another PR to your branch to tweak the docs a bit to match our general style a little closer.

Reply:
I have moved the exports into a separate file now as discussed. I also added an explanatory comment to the file to clarify its purpose, even including a simple usage example.